test/e2e/utils_test.go

```diff
@@ -66,39 +79,63 @@ func verifyOperatorStatusCondition(client *certmanoperatorclient.Clientset, cont
 	return errors.NewAggregate(errs)
 }
 
-// removeOverrides removes all the overrides from all the cert-manager operands. The update process is retried if
-// a conflict error is encountered.
-func removeOverrides(client *certmanoperatorclient.Clientset) error {
-	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		operator, err := client.OperatorV1alpha1().CertManagers().Get(context.TODO(), "cluster", v1.GetOptions{})
+// resetCertManagerState is used to revert back to the default cert-manager operands' state
+func resetCertManagerState(ctx context.Context, client *certmanoperatorclient.Clientset, loader library.DynamicResourceLoader) error {
+	// update operator spec to empty *Config and set operatorSpec to default values
+	err := wait.PollImmediate(PollInterval, TestTimeout, func() (bool, error) {
+		_, err := client.OperatorV1alpha1().CertManagers().Get(ctx, "cluster", v1.GetOptions{})
 		if err != nil {
-			return err
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
 		}
 
-		updatedOperator := operator.DeepCopy()
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			operator, err := client.OperatorV1alpha1().CertManagers().Get(ctx, "cluster", v1.GetOptions{})
+			if err != nil {
+				return err
+			}
```

Comment:
It would make more sense if the `wait.PollImmediate` is called inside `retry.RetryOnConflict` to get the most recent cert-manager object. Only the get call can be added inside `wait.PollImmediate`, and once the object is fetched the other actions can be performed on it, including the update.

Reply:
Understood, let me update.

Reply:
Done.